### DISCUSSION TITLE
How to replace some regex in all cells?

### DISCUSSION THREAD

**Discussion by halloleo** [saulpw/visidata#2502] [Q&A] [answered]
How can I replace in all cells of a VisiData document a text which matches a given regex ? (In my particular case I want to remove all non-number text *incl. in-cell carriage returns*.)

_As an entertaining aside:_ I asked this [Perplexity.ai](https://www.perplexity.ai) And it gave me something which might be close or not: Typing `g*` which apparently executes the command  "global-substitution". However `g*` didn't work for me - and the command "global-substitution" I could not find on VisiData's commands sheet. -- It is my experience with ChatGPT-like agents in regards to VisiData, that they are often pretty far off the mark. Maybe we would need to feed the AI the whole VisiData doco in a custom LLM, but that's at this stage beyond my abilities.)

**Comment by saulpw** [comment]
There isn't a command that can do this, currently. But I think the `g*` command could be hacked up pretty easily into a new command that did the same thing but over all columns and rows (instead of just the selected rows in the current column).

**Comment by midichef** [comment, marked as answer]
There's no single command do this, but you can do `gs` to select all rows, then `gz*`, which does `setcol-regex-subst-all`. However, it seems it will not replace text within dates (in a column of data that has the `date` type, showing the `@` symbol).

**Reply by saulpw** [reply]
I forgot about `gz*`, good find :)

**Reply by halloleo** [reply]
Super cool find indeed. Works perfectly. Thank you. -- How can I find a command combo myself?

**Reply by midichef** [reply]
I looked at the HelpSheet of commands for the current sheet. Then I looked for `subst`, since I knew `*` (`addcol-regex-subst`) had that action in its name. The keypresses for that were:  `zCtrl+H` `c` `longname` `|` `subst`.

**Reply by halloleo** [reply]
Super interesting. Makes total sense when you know `*` and its command name.

I use `vd` only every couple of weeks -- and for occasional users like me the command _discovery_ seems to be _a bit_ hard...
